mllib/src/main/scala/org/apache/spark/ml/feature/QuantileDiscretizer.scala

```diff
@@ -49,6 +49,21 @@ private[feature] trait QuantileDiscretizerBase extends Params
 
   /** @group getParam */
   def getNumBuckets: Int = getOrDefault(numBuckets)
+
+  /**
+   * Relative error (see approxQuantile documentation for description).  Must be >= 0.
+   * By default, the relative error will be in the range [0.001, 0.01]. The exact value
+   * is computed based on the number of buckets, decreasing as the number of buckets
+   * increases. Explicitly setting the relative error will override this default calculation.
+   * @group param
+   */
+  val relativeError = new DoubleParam(this, "relativeError", "The relative target precision " +
+    "for approxQuantile",
+    ParamValidators.gtEq(0.0))
```

Comment:
Just occurred to me, the error cannot be > 1.0 correct? I know `approxQuantile` accepts > 1 (which has same behaviour as 1.0), but to me it makes sense to explicitly restrict it here to `[0, 1]` - you can use `ParamValidators.inRange` for this.